components/brave_today/common/brave_news.mojom

```diff
@@ -0,0 +1,133 @@
+module brave_news.mojom;
+
+import "mojo/public/mojom/base/time.mojom";
+import "url/mojom/url.mojom";
+
+// Items either have a padded or an unpadded image url
+union Image {
+  url.mojom.Url padded_image_url;
+  url.mojom.Url image_url;
+};
+
+enum CardType {
+  HEADLINE,
+  HEADLINE_PAIRED,
+  CATEGORY_GROUP,
+  PUBLISHER_GROUP,
+  DEALS,
+  DISPLAY_AD,
+  PROMOTED_ARTICLE
+};
+
+struct FeedItemMetadata {
+  string category_name;
+  mojo_base.mojom.Time publish_time;
+  string title;
+  string description;
+  url.mojom.Url url;
+  string url_hash;
+  Image image;
+  string publisher_id;
+  string publisher_name;
+  double score;
+  string relative_time_description;
+};
+
+struct Article {
+  FeedItemMetadata data;
+};
+
+struct PromotedArticle {
+  FeedItemMetadata data;
+  string creative_instance_id;
+};
+
+struct Deal {
+  FeedItemMetadata data;
+  string offers_category;
+};
+
+union FeedItem {
+  Article article;
+  PromotedArticle promoted_article;
+  Deal deal;
+};
+
+struct FeedPageItem {
+  CardType card_type;
+  // Each UI must validate if the items array
+  // contains enough content to support the card_type.
+  array<FeedItem> items;
+};
+
+struct FeedPage {
+  array<FeedPageItem> items;
+};
+
+struct Feed {
+  string hash;
+  array<FeedPage> pages;
+  FeedItem? featured_item;
+};
+
+enum UserEnabled {
+  NOT_MODIFIED,
+  ENABLED,
+  DISABLED,
+};
+
+struct Publisher {
+  // Identifies whether the user has manually set the publisher on and off,
+  // so that we may know if the source should follow the remote-provided value
+  // (which may change), or not.
+  string publisher_id;
+  string publisher_name;
+  string category_name;
+  bool is_enabled;
+  UserEnabled user_enabled_status = UserEnabled.NOT_MODIFIED;
+};
+
+struct DisplayAd {
+  string uuid;
+  string creative_instance_id;
+  string dimensions;
+  string title;
+  string description;
+  Image image;
+  url.mojom.Url target_url;
+  string? cta_text;
+};
+
+// Browser-side handler
+interface BraveNewsController {
+  GetFeed() => (Feed feed);
+  GetPublishers() => (map<string, Publisher> publishers);
+  GetImageData(url.mojom.Url padded_image_url) => (array<uint8>? image_data);
+  SetPublisherPref(string publisher_id, UserEnabled new_status);
+  ClearPrefs();
+  IsFeedUpdateAvailable(string displayed_feed_hash) => (bool is_update_available);
```

Comment:
Can we please ensure that lines a shorter than 80 and also throw in some empty lines between methods, especially commented ones?

Reply:
hmm, yes on the 80 chars. I would argue against the usefulness of empty lines, instead preferring empty lines between code blocks, and I like the style of using lines of comments to separate lines of code that require explanation. Unless we have a style guide that contradicts that?

Reply:
this is in Google C++ style guide https://google.github.io/styleguide/cppguide.html#Vertical_Whitespace
```
A blank line before a comment line usually helps readability — the introduction of a new comment suggests the start of a new thought, and the blank line makes it clear that the comment goes with the following thing instead of the preceding.
```
Also this is how most of existing code looks, actually.